### PR TITLE
JMH bench and flight recorder for reliable delivery, #28725

### PR DESCRIPTION
**Pull request by patriknw**
The results strengthen my suspicion that the unstashing in ConsumerController #28718 is important.

References #28725

### CODE REVIEW

File: akka-actor-typed/src/main/scala-jdk-9/akka/actor/typed/internal/jfr/Events.scala [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+@InternalApi
+@Enabled(true)
+@StackTrace(false)
+@Category(Array("Akka", "Delivery", "ConsumerController")) @Label("Delivery ConsumerController started")
+final class DeliveryConsumerStarted(val actorPath: String) extends Event
+
+/** INTERNAL API */
+@InternalApi
+// FIXME false
+@Enabled(true) // hi frequency event
```
Comment (patriknw) [inline akka-actor-typed/src/main/scala-jdk-9/akka/actor/typed/internal/jfr/Events.scala]: before merging I should disable the hi frequency events by default

**Review by johanandren — APPROVED** [review]
LGTM (with that hi freq disabled by default)

**Review by chbatey — APPROVED** [review]

**Comment by patriknw** [comment]
disabled hi freq

**Comment by patriknw** [comment]
For reference, to run the bench with JFR enabled I used:
```
jmh:run -jvmArgsAppend -XX:+FlightRecorder -jvmArgsAppend -XX:+UnlockDiagnosticVMOptions -jvmArgsAppend -XX:+DebugNonSafepoints -jvmArgsAppend -XX:StartFlightRecording=filename=deliveryrecording.jfr,settings=profile,path-to-gc-roots=true,maxsize=1024m -i 5 -wi 2 -f1 -t1 akka.actor.typed.delivery.ReliableDeliveryBenchmark
```

**Comment by patriknw** [comment]
PLS BUILD

**Comment by patriknw** [comment]
There was one strange thing with the build. Ordinary test:compile didn't compile JDK9 classes, resulting in

```
ClassNotFoundException: akka.actor.typed.internal.jfr.JFRActorFlightRecorder
```

It's done by `package`, so triggered by jmh-bench.

Added some `actorTyped % "compile->CompileJdk9"` in 2b0e856. That works but I'm not quite sure that it's the right way. I was expecting that `actorTyped` itself would CompileJdk9 as part of its compile, but that is not the case.

Same problem with Artery JFR classes.

@raboof @johanandren  Any thoughts?

**Comment by raboof** [comment]
> Added some `actorTyped % "compile->CompileJdk9"` in 2b0e856. That works but I'm not quite sure that it's the right way.

Normally this would be problematic at compile time when only on jdk8, because then the classes in `CompileJdk9` are not available. That's not a problem here because `compile` only refers to the `CompileJdk9` classes reflectively.

Another option would be to remove this dependency, and replace the `JavaVersion.majorVersion >= 11` condition with a check whether that class is available on the classpath. Then any classes that want to exercise the Jdk9+ features would have to be placed in `TestJdk9`.

I think this would make the build a bit simpler (aligning the build-time and run-time dependencies), though it'd also mean the flight recorder doesn't get as much exercise while running non-jdk9-only tests.

Perhaps it is OK to keep it like this, but we should start considering dropping support for building Akka with JDK8? (of course we'll still support running Akka on JDK8).

**Review by raboof — APPROVED** [review]
Changes LGTM, should/can we have a test that tests anything is actually recorded or is that only observable with the mission control software?

**Comment by patriknw** [comment]
> can we have a test that tests anything is actually recorded

I think they added some API for streaming events in JDK14, but at the moment that's probably difficult.

**Comment by raboof** [comment]
> Another option would be to remove this dependency, and replace the JavaVersion.majorVersion >= 11 condition with a check whether that class is available on the classpath. Then any classes that want to exercise the Jdk9+ features would have to be placed in TestJdk9.

I think we should do this to prevent accidentally adding any code later where 'regular' code depends on classes that are not available jdk9+ and thus would lead to runtime errors when running on jdk8. I'll file a separate ticket.

**Comment by jrudolph** [comment]
It seems this change broke the akka-bench-jmh project (#28889). Hard to say what's wrong exactly, could we try to revert the broken part for the time being?

**Pull request merged**